Developer documentation overhaul

taldcroft opened the pull request:
<!-- These comments are hidden when you submit the pull request,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/main/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/main/CODE_OF_CONDUCT.md . -->

<!-- If you are new or need to be re-acquainted with Astropy
contributing workflow, please see
http://docs.astropy.org/en/latest/development/workflow/development_workflow.html .
There is even a practical example at
https://docs.astropy.org/en/latest/development/workflow/git_edit_workflow_examples.html#astropy-fix-example . -->

<!-- Please just have a quick search on GitHub to see if a similar
pull request has already been posted.
We have old closed pull requests that might provide useful code or ideas
that directly tie in with your pull request. -->

<!-- We have several automatic features that run when a pull request is open.
They can appear daunting but do not worry because maintainers will help
you navigate them, if necessary. -->

### Description
<!-- Provide a general description of what your pull request does.
Complete the following sentence and add relevant details as you see fit. -->

<!-- In addition please ensure that the pull request title is descriptive
and allows maintainers to infer the applicable subpackage(s). -->

<!-- READ THIS FOR MANUAL BACKPORT FROM A MAINTAINER:
Apply "skip-basebranch-check" label **before** you open the PR! -->

This pull request is to make the developer documentation more accessible to newcomers and more useable by  maintainers. The current docs contain a lot of information but are not well-organized and contain a lot of repetition.

It reflects an approach that sometimes "less is more", so some material was cut (notably detailed documentation on how to use two virtualenv managers), but for the most part it keeps all substantive material while cutting repetitions.

There are a lot of commits and most of them are useless individually, so this should definitely be squash-merged. There are a couple of commits that could be controversial and are meaningful.

<!-- If the pull request closes any open issues you can add this.
If you replace <Issue Number> with a number, GitHub will automatically link it.
If this pull request is unrelated to any issues, please remove
the following line. -->

Fixes #11621

See also:
- https://github.com/astropy/astropy/pull/11207
- https://github.com/astropy/astropy/pull/16208
- https://github.com/astropy/astropy/pull/16555

<!-- Optional opt-out -->

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

neutrinoceros commented:
> Is there an issue for this??

I found two related issues:
- https://github.com/astropy/astropy/issues/11621
- https://github.com/astropy/astropy/issues/11207
From your description, it sounds like this PR would maybe close the second one as "not planned" ?

taldcroft commented:
RTD is now passing so you can look at the new docs.

hamogu commented:
I can't put a comment on it because this line is not yet changed, but I would also do away with `astropy.test` in favor of recommending to run pytest directly (https://github.com/astropy/astropy/pull/16561/files#diff-3082c98617b79c67603d9ef61bad41ef94d24d19a948af26eb52ca8ace968a97L368). Keep it simple by recommending one and only one way of doing it and we have #16208 open to remove `astropy.test()` anyway.

taldcroft commented:
> I can't put a comment on it because this line is not yet changed, but I would also do away with astropy.test in favor of recommending to run pytest directly

I had not appreciated this was technically possible for an installed package. But definitely :+1: on removing `astropy.test()` from the docs, will do (in coordination with #16208).

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+If you have not done so already, you will need your own copy of ``astropy`` to
+build it and/or contribute to the source. Astropy is hosted in the `astropy GitHub repository <https://www.github.com/astropy/astropy>`_ and you need to make a clone.
+
+If you *only want to build astropy* but not contribute, you can run the following and then skip to the :ref:`next section<create-isolated-env>`::
```
I wonder if we should not omit this too - i.e., keep this the quickstart guide to making a contribution.

taldcroft replied:
This came out of linking to this bit (so re-using these instructions) from the Install page, where you might want to only build the docs or run tests. I also didn't want to make more branches here but otherwise the original Install had independent instructions.

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+    git clone https://github.com/YOUR-USER-NAME/astropy.git
+    cd astropy
+    git remote add upstream https://github.com/astropy/astropy.git
+    git fetch upstream
```
`git fetch upstream` is not really necessary for those who just forked.

taldcroft replied:
That's what I thought until I tried it... It pulls down all the tags. But I'm fine removing it.

mhvk replied:
Ah, in that case perhaps leave it...

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+for doing other work. This is important because the development environment will often
+be unstable and possibly broken at times, and you don't want to break your other work.
+
+There are many good options for doing this, including a number of virtual environment
```
This is a matter of choice, so feel free to ignore, but since we use `tox` for testing, one could suggest using it here too. (Of course, this just so happens to be what I do these days -- `tox -e test --develop --notest`.)

taldcroft replied:
Conda does two things that tox does not (AFAIK):
1. Easily allow installing `graphviz` so that the docs build without hundreds of long warnings that completely obscure real warnings/errors.
2. Easily allow installing different Python versions.

Using conda also teaches people about working in isolated environments so that hopefully this transfers to other domains. Although tox makes a virtual env, using it in the context of testing hides the little bit of environment management that we document and use here. So overall IMO we should just settle on conda as the base environment manager for astropy.

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+Running the full test suite can take a while, so you may want to start with a subset
+of only the coordinates tests::
+
+.. code-block:: shell
```
This is not rendered correctly - maybe a problem with the double :: just above?

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+
+.. _contributing.version_control:
+
+GitHub and Git
```
This feels like a bit too much duplication from the start of the page.  I'd skip directly to creating a feature branch

taldcroft replied:
Agreed, fixed.

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+.. Important:: Never use the ``main`` branch for making changes. Always create a new
+   feature branch for your changes.
+
+When you want to update the feature branch with changes in main after
```
This feels slightly out of place here, and you have a more useful warning further down. So perhaps remove here? Or move the important point here.

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+
+
+
+You can see the remote repositories
```
This might be better at the end of the part where `upstream` is added.

taldcroft replied:
Done

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+with a useful link to create a pull request:
+
+  remote: Create a pull request for 'subpackage-new-feature' on GitHub by visiting:
+  remote:      https://github.com/yourname/astropy/pull/new/subpackage-new-feature
```
This gets typeset funnily at readthedocs - at least, I usually see exactly what you have here, but on readthedocs, it replaces the github part by an emoji

taldcroft replied:
Missing a double `::`, should be fixed now.

mhvk commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+Most of the time you can just follow the link that ``git`` provide when you pushed
+your branch and create the PR. If you don't have that link (and for a few more details), you can follow the :ref:`pull-request` instructions.
+
+Follow the instructions in the PR template and fill it out as completely as possible.
```
Should this say something about labels? Or can normal users not set those?

hamogu replied:
Every user can **see** labels, but only maintainers can **set** them.

mhvk reviewed (commented):
@taldcroft - so far, I've just gone over the new quickstart guide - which I like very much! But of course some comments (mostly small).

taldcroft commented on docs/development/testguide.rst:
```diff
@@ -6,68 +6,49 @@
+
+    python -m pip install --editable ".[dev_all]"
+
+This will provide all dependencies for running the full test suite using ``tox``,
+``pytest`` and ``astropy.test()``. It will also allow running tests via any IDE which
```
Done.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+.. code-block:: shell
+
+    git add files-that-you-changed ...
+    git commit -m "your commit message goes here"
```
Why? I've used that successfully for over a decade without any problems. GitHub seems to think it is fine: https://docs.github.com/en/get-started/using-git/about-git#example-contribute-to-an-existing-repository

eerovaher replied:
A short answer is that the level of detail in commit messages sets an upper limit to how useful `git log` and `git blame` can be, but you can have a look at [Git's own guidelines](https://git-scm.com/docs/SubmittingPatches#describe-changes) for a longer  discussion.

taldcroft replied:
From the git guidelines "those who need to fix or enhance your code in the future will need to know *why* your code does what it does". I could not agree more, except that the *why* should be in code comments, not in a commit message.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+    git add files-that-you-changed ...
+    git commit -m "your commit message goes here"
+
+You should make frequent commits and always include a commit message. Each commit
```
I added the link, but I don't agree about spending a great deal of time on commit messages. I pretty much never read commit messages and instead I focus more on the code being well documented so will make sense in 10 years (even when git blame no longer points to the original commit).

eerovaher replied:
Writing good commit messages should not take long if the commits are sensibly organized.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+When you want your changes to appear publicly on your GitHub page, push your
+forked feature branch's commits::
+
+    git push origin subpackage-new-feature
```
To be honest I am sometimes confused by what the defaults are, and in general I won't remember if I have set an upstream them in any of the dozens of projects I'm working on. So my practice is just always being explicit. 

I *think* that in the instructions outlined here (with somewhat recent `git`), one could use the minimal command `git push` or `git push subpackage-new-feature`. Would you prefer either of those?

eerovaher replied:
Without `--set-upstream` (or some other way of adding a reference to the remote branch) the remote needs to be specified on every push, but with `--set-upstream` it needs to be written out only for the first push.

taldcroft replied:
I added `--set-upstream`.

taldcroft commented on docs/development/testguide.rst:
```diff
@@ -82,37 +63,23 @@ issues related to undeclared package data, or missing dependencies. Since we use
-will run checks using the ``ruff`` tool.
+    tox -e test  # Run the tests with the minimal set of dependencies
+    tox -l -v  # Print a description of all available test environments
+    tox -e codestyle  # Run code style checks using ``ruff``
```
I didn't write this, but it just highlights another tox option which is guaranteed (I think?) to match what is done on CI.

eerovaher replied:
The `tox` codestyle job just calls `pre-commit` with extra steps: https://github.com/astropy/astropy/blob/62018574e74cda9c4b30b088f4049e01fb403fbd/tox.ini#L149-L156

I wouldn't mention `tox` in a basic contributor guide at all.

taldcroft replied:
> I wouldn't mention tox in a basic contributor guide at all.

This is in the testing guide, which is meant to be somewhat cover all recommended testing options. Tox falls in that category, and some developers strongly prefer tox.

taldcroft commented on docs/development/testguide.rst:
```diff
@@ -138,47 +105,29 @@ pytest
-
-Or, to run only the ``wcs`` and ``utils`` tests::
-
     pytest -P wcs,utils
```
In my own use I agree, but this ignores Windows developers.

eerovaher replied:
@pllim, is it true you can't do brace expansion on Windows?

pllim replied:
I never even knew about brace expansion. I will have to check when I get a chance. 😅

pllim replied:
Re: braces on Windows -- Works on Windows 11 but I use Git Bash within VS Code, not Powershell. @duytnguyendtn , do you still use Powershell? Would you mind running this using code checkout and installed from `main` (does not have to be this PR branch):

```
pytest {astropy,docs}/{wcs,utils} -p no:cacheprovider
```

I had to give `-p no:cacheprovider` because of Norton but if you don't have the cache dir creation problem, you don't need this option.

Thanks!

duytnguyendtn replied:
Indeed looks like brace expansion isn't supported in Powershell, or at least there's something syntactically off; it didn't even execute:

```
(envastropypr16561) PS C:\Users\dtnguy20\Documents\duydir\gitrepos> pytest {astropy,docs}/{wcs,utils}
At line:1 char:16
+ pytest {astropy,docs}/{wcs,utils}
+                ~
Missing argument in parameter list.
At line:1 char:27
+ pytest {astropy,docs}/{wcs,utils}
+                           ~
Missing argument in parameter list.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingArgument
```

duytnguyendtn replied:
There's a whole thread on StackOverflow on people trying to find hacky workarounds: https://stackoverflow.com/questions/16222477/powershell-equivalent-of-bash-brace-expansion-for-generating-lists-arrays

pllim replied:
Wow, glad I asked. Thanks for checking and the extra info!

@eerovaher , I guess we cannot accept your suggestion in using brace expansion then, at least for now.

eerovaher replied:
I am still convinced that my suggestion was sensible, but I can accept that Powershell is not.

pllim replied:
I do not disagree with you when it comes to Powershell and sensibility. But we're stuck in the universe where it exists...

duytnguyendtn replied:
I can only apologize for the behavior of the windows overlords... Thank you for not forgetting about us Windows devs!

On the other hand, Powershell *is* open-source 👀: https://github.com/PowerShell/PowerShell

taldcroft commented:
Thanks for the comments folks! I got through many but not all of them, but I'll keep on plugging (hopefully over the next few days).

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+    git add files-that-you-changed ...
+    git commit -m "your commit message goes here"
+
+You should make frequent commits and always include a commit message. Each commit
+should represent one logical set of changes.
```
Good point. This is a bit subtle and complex, and feels out-of-scope at this level. I just removed it since our quickstart guide is not going to teach people how to be effective developers. Instead just get them through a PR and they will start to learn with feedback from maintainers.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+Follow the instructions in the PR template and fill it out as completely as possible.
+
+If your PR is still a work in progress then instead of clicking "Create pull request",
+click on the small down arrow next to it and select "Create draft pull request".
```
Done.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+
+If your PR is still a work in progress then instead of clicking "Create pull request",
+click on the small down arrow next to it and select "Create draft pull request".
+In addition, if your commits are not ready for CI testing, you
+should include ``[ci skip]`` the last commit message.
```
Done.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+
+- **Reference an open issue** for non-trivial changes to clarify the PR's purpose.
+- **Ensure you have appropriate tests**.
+- **Keep your pull requests as simple as possible** -- larger PRs take longer to review.
```
Good point, done.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+maintainers will take a look. To make the process as smooth and efficient as possible,
+here are some tips:
+
+- **Reference an open issue** for non-trivial changes to clarify the PR's purpose.
```
Agreed and done.

taldcroft commented:
All -- I think that I have addressed or at least responded to every comment. I think this is in good enough shape now that I have made it a Pull Request (not draft).

neutrinoceros commented:
I'm going to review this today. For now, I just want to put a note that #16601 revealed an oversight in our dev docs. Fetching tags is necessary to first build and test astropy locally, and [the information is there](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#fetch-the-latest-astropy), but isn't presented as something you should do from the get go when you're getting started. Maybe you've already addressed this point (in which case I'll come back and hide this reply), but I wanted to make sure we addressed it too as part of the overhaul.

neutrinoceros reviewed (commented):
Some general level questions: 
1) why were some files from `docs/development/workflow` moved, but not all ?
2) why replacing `git checkout` with `git switch` everywhere ? (this is a genuine question, I learned git before `git switch` was a thing and never really bothered migrating my muscle memory)

eerovaher commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+    git clone https://github.com/YOUR-USER-NAME/astropy.git
+    cd astropy
+    git remote add upstream https://github.com/astropy/astropy.git
+    git fetch upstream
```
I see the value in making similar recommendations in a quickstart guide as the GitHub documentation does, but our documentation should clearly state that this is what we are doing.

taldcroft replied:
Will do.

eerovaher commented on docs/development/testguide.rst:
```diff
@@ -6,68 +6,49 @@
+Details
+=======
 The dependencies used by the Astropy test runner are provided by a separate
-package called |pytest-astropy|. This package provides the ``pytest``
+package called |pytest-astropy| (see :ref:`pytest-plugins`). This package provides the ``pytest``
 dependency itself, in addition to several ``pytest`` plugins that are used by
 Astropy, and will also be of general use to other packages.
```
To me it is obvious that `astropy` contributor guidelines and affiliated package maintainer instructions should be two separate documents.

taldcroft replied:
OK, moved this into `maintainers/testhelpers.rst`.

pllim commented:
@taldcroft , looks like there is conflict in `docs/install.rst` now. Please rebase. Thanks!

taldcroft commented on docs/development/maintainers/maintaining.rst:
```diff
@@ -0,0 +1,469 @@
+
+.. code-block:: python
+
+    import astropy as pd
```
@neutrinoceros - this is a leftover from pandas where I copied over the pandas dev docs and changed pandas to astropy but didn't do any other editing. Sorry for any confusion!

In order to avoid further confusion, I'm going to remove that `maintaining.rst` file but folks should look at https://pandas.pydata.org/docs/development/maintaining.html to see if there is anything astropy could be doing differently or better.

taldcroft commented:
@pllim - rebased!

pllim commented:
I applied Extra CI label for linkcheck. Can cancel the rest (e.g., exotic archs) if you want.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,365 @@
+module). Users who have a preference for a particular virtual environment manager are
+encouraged to use it!
+
+For this quickstart guide we use the `conda <https://docs.conda.io/en/latest/>`_ package
```
#16555 is for installing a release version of astropy. This PR deals with installing astropy from source with the intent of developing code or docs, running tests etc. Overall, this PR is a huge change and if we have a long discussion about every small detail it will never converge.

pllim reviewed (commented):
Given the sheer amount of diff and contents, I couldn't carefully read through everything. But overall, this is a great update! Once comments are addressed, I think we can merge and do smaller improvements from there. Thanks!

**However,** my review here is incomplete but I am posting now because I have to step away and don't want to lose what I have. I plan to come back to this PR and post the rest of my review before the end of the week.

pllim commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,331 @@
+be unstable and possibly broken at times, and you don't want to break your other work.
+
+There are many good options for doing this, including a number of virtual environment
+managers (e.g. the Python standard library `venv <https://docs.python.org/3/library/venv.html>`_
```
This is also a general comment that applies to other occurrences. According to https://astropy--16561.org.readthedocs.build/en/16561/development/style-guide.html#abbreviations

"Place abbreviations such as i.e. and e.g. within parentheses, where they are followed by a comma."

mhvk commented on docs/development/codeguide.rst:
```diff
@@ -188,8 +188,6 @@ Coding Style/Conventions
 
   .. note:: There are multiple options for testing PEP8 compliance of code,
             see :doc:`testguide` for more information.
-            See :doc:`codeguide_emacs` for some configuration options for Emacs
```
Me too... :crying_cat_face:

taldcroft replied:
Cleaning up old stuff is not always easy.

pllim commented on docs/development/codeguide.rst:
Also not in the diff and maybe @dhomeier or @hamogu can advise: With pyOpenSci collaboration, https://astropy--16561.org.readthedocs.build/en/16561/development/codeguide.html#requirements-specific-to-affiliated-packages seems out of place and incomplete now. Should we gut out the content and point to https://www.astropy.org/affiliated/#becoming-an-affiliated-package instead?

hamogu replied:
I agree with @taldcroft : Let's get this one merged and then we'll see.

pllim replied:
https://github.com/astropy/astropy/issues/16624

taldcroft commented on docs/development/development_details.rst:
```diff
@@ -666,6 +356,47 @@ can delete any backup branches that may have been created::
+
+.. _external_c_libraries:
+
+External C Libraries
```
No idea. This PR is not intended as an audit of every detail in the dev docs, but more a refactor, simplification, and elimination of redundant content.

pllim replied:
https://github.com/astropy/astropy/issues/16623

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,331 @@
+code from your local repository becomes available to the GitHub community to review and
+merged into project to appear the in the next release.
+
+Most of the time you can just follow the link that ``git`` provide when you pushed
```
I mean the link below:
```
(astropy) ➜  docs git:(doc-dev-quickstart) git push origin github-pr-link 
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote: 
remote: Create a pull request for 'github-pr-link' on GitHub by visiting:
remote:      https://github.com/taldcroft/astropy/pull/new/github-pr-link
remote: 
To https://github.com/taldcroft/astropy.git
 * [new branch]            github-pr-link -> github-pr-link
```

taldcroft commented:
> * why were some files from `docs/development/workflow` moved, but not all ?

Maybe this is overtaken? I don't see any files there still.

> * why replacing `git checkout` with `git switch` everywhere ? (this is a genuine question, I learned git before `git switch` was a thing and never really bothered migrating my muscle memory)

`git switch` (and `git restore`) were introduced because `get checkout` is so overloaded and confusing.

taldcroft commented:
OK, I fixed or responded to all of the comments as far as I can see. I hope this is getting close...

neutrinoceros commented:
@taldcroft you'll also need to rebase (or invoke @pllim's magic) to get this merged, as some currently required jobs are not running at all.

taldcroft commented:
I added What's New and changelog entries for this and rebased.

dhomeier commented:
> > * why replacing `git checkout` with `git switch` everywhere ? (this is a genuine question, I learned git before `git switch` was a thing and never really bothered migrating my muscle memory)
> 
> `git switch` (and `git restore`) were introduced because `get checkout` is so overloaded and confusing. It took some effort but I got `git switch` into my fingers.

Had not even heard of it before. I think I like the new interface better, but would like to note
https://git-scm.com/docs/git-switch#_description

pllim commented:
> Had not even heard of it before.

At some point, Git decided that some commands were too overloaded and confusing, so they introduced new commands. But with my muscle memory set, I never bothered. `git checkout` works just as well, but this is indeed clearer for new people.

pllim commented:
```
whatsnew/7.0 404 Client Error: Not Found for url: https://docs.astropy.org/en/latest/development/
whatsnew/7.0 404 Client Error: Not Found for url: https://docs.astropy.org/en/latest/development/quickstart.html
```

But these work, so I guess we will just have to ignore the broken links for now.

https://astropy--16561.org.readthedocs.build/en/16561/development/
https://astropy--16561.org.readthedocs.build/en/16561/development/quickstart.html

dhomeier commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,335 @@
+    pre-commit install
+
+Now all of the styling checks will be
+run each time you commit changes without your needing to run each one manually.
```
```suggestion
run each time you commit changes without your needing to run each one manually.
These same checks will also be run on GitHub once you have made or updated your pull request
(see below) and need to pass before your changes can be accepted.
```
I still managed on first pass to overread the pre-commit section here, so can only help to give it more visibility, I hope.

pllim replied:
I propose you replace the suggestion of "(see below)" with explicit intersphinx to be less vague and more future-proof of any further re-org.

dhomeier replied:
This will depend a bit on the pre-commit.ci reference being added there.

taldcroft replied:
Addressed in [1e42733](https://github.com/astropy/astropy/pull/16561/commits/1e427332baca08db47bf10c452802732f6a69682) with slight changes.

dhomeier commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,335 @@
+If your PR is still a work in progress then instead of clicking "Create pull request",
+click on the small down arrow next to it and select "`Create draft pull request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests>`__".
+In addition, if your commits are not ready for CI testing, you
+should include ``[ci skip]`` the last commit message (but note that code formatting
+checks and documentation building will still be done).
```
```suggestion
should include ``[ci skip]`` the last commit message – but note that code formatting
checks and documentation building will still be done. Formatting and style error *should*
already have been fixed before committing if you did :ref:`_contributing.pre-commit`,
but if you have not, you can use the :ref:`pre-commit_bot` to fix them automatically in the PR.
```
Sorry, my RST skills are very lame; I just hope that the first reference points to the quick intro inside this file.

pllim replied:
I don't think you should include the leading underscore.

```rst
:ref:`contributing.pre-commit`
```

dhomeier reviewed (approved):
Thanks for undertaking this, this should point new contributors to a much clearer path!
I just had time for reading through the quickstart and core workflow sections, only a few suggestions for potential improvements.

pllim commented on docs/index.rst:
```diff
@@ -37,7 +37,7 @@ processing, and data analysis.
    index_getting_started
    index_user_docs
-   index_dev
+   development/index
```
I think there was a reason why we made the index at top level (`index_dev`) vs `development/index` but I don't remember why now. @saimn , do you remember?

pllim replied:
Just going to xref https://github.com/astropy/astropy/pull/14867 here in case we need to revert this particular change in the future.

pllim commented on docs/development/index.rst:
```diff
@@ -0,0 +1,79 @@
+outlined here and `ask for help <https://www.astropy.org/help.html>`_ if you get stuck.
+The Astropy community is welcoming and friendly and will help you!
+
+This is divided into two sections, first a quickstart guide that provides an
```
Should this paragraph also be moved under `{% if is_development %}`? Seems weird to explain there are two sections when in stable, it would ask you to go to latest anyway.

taldcroft replied:
Fixed.

pllim commented on docs/install.rst:
```diff
@@ -247,138 +228,41 @@ The following packages can optionally be used when testing:
-installing ``astropy``.
+Building the documentation is typically not necessary unless you are
+developing code or documentation or do not have internet access, because
+the latest (and archive) versions of Astropy's documentation are
```
```suggestion
the latest (and archived) versions of Astropy's documentation are
```

taldcroft replied:
Fixed separately.

pllim commented on docs/install.rst:
```diff
@@ -247,138 +228,41 @@ The following packages can optionally be used when testing:
+Building the documentation is typically not necessary unless you are
+developing code or documentation or do not have internet access, because
+the latest (and archive) versions of Astropy's documentation are
+available at `docs.astropy.org <https://docs.astropy.org>`_ . The process
```
This needs clarification because https://docs.astropy.org routes to stable doc so it is very confusing.

taldcroft replied:
Updated to:
```
... because the stable, latest, and archive versions of Astropy's documentation are
available at `docs.astropy.org <https://docs.astropy.org>`_ .
```

pllim commented on docs/known_issues.rst:
Are you sure we can remove these known issues? I think in the RC testing, we still ask people to run `astropy.test()`.

taldcroft replied:
Not any more, we tell them to use`pytest --pyargs astropy`, documented in the testing guide. I added a link to the instructions for that in `Installing pre-built Development Versions of astropy`.

pllim commented on docs/whatsnew/7.0.rst:
Hmm why is no one populating the TOC of this What's New page? I guess a follow-up PR is warranted.

pllim replied:
https://github.com/astropy/astropy/issues/16626

taldcroft replied:
Hah, I saw the precedent and thought maybe that is a new thing and I should just follow it.

pllim reviewed (commented):
Here is the rest of my review. Since you said contents update largely is out of scope, I focused only on the stuff not inside `development` this time.

taldcroft commented on docs/development/quickstart.rst:
```diff
@@ -0,0 +1,335 @@
+
+**MacOS**
+
+Install the Developer Tools using ``xcode-select --install``.
```
Addressed in [1e42733](https://github.com/astropy/astropy/pull/16561/commits/1e427332baca08db47bf10c452802732f6a69682) with slight changes.

taldcroft commented:
OK, I think I have addressed all review comments. There are some other PR's being blocked by this, so maybe it is time to merge this and start fresh with any other minor cleanups?

taldcroft commented:
@pllim - are you OK with merging now? I see 👍 from a few others that have reviewed.

pllim commented:
There is an edited comment from @dhomeier saying that `git switch` is experimental at https://github.com/astropy/astropy/pull/16561#issuecomment-2194728046 . Should we be recommending experimental Git command?

pllim commented:
devdeps failures are unrelated.

taldcroft commented:
> There is an edited comment from @dhomeier saying that `git switch` is experimental at [#16561 (comment)](https://github.com/astropy/astropy/pull/16561#issuecomment-2194728046) . Should we be recommending experimental Git command?

Git switch was released in 2.23 almost 5 years ago. I'm not sure why they have not removed the "experimental" comment, but I don't think it is credible that they would now drop or make big changes to `git switch`.

pllim commented:
Oh... ok then. Are you okay with "Squash and Merge"? 🚀

taldcroft commented:
> Oh... ok then. Are you okay with "Squash and Merge"? 🚀

For sure squash and merge. There is no value in all the intermediate commits and LOTS of churn!

pllim commented:
Done. Thanks!!!

mhvk commented:
Thanks! So nice to have clearer devdocs!!
